core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -304,18 +304,20 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
   }
 
   private def validateKillArguments(): Unit = {
-    if (!master.startsWith("spark://") && !master.startsWith("mesos://")) {
-      error("Killing submissions is only supported in standalone or Mesos mode!")
+    if (!master.startsWith("spark://") && !master.startsWith("mesos://")
+      && !master.startsWith("k8s://")) {
+      error("Killing submissions is only supported in standalone, Mesos or K8s mode!")
     }
     if (submissionToKill == null) {
       error("Please specify a submission to kill.")
     }
   }
 
   private def validateStatusRequestArguments(): Unit = {
-    if (!master.startsWith("spark://") && !master.startsWith("mesos://")) {
+    if (!master.startsWith("spark://") && !master.startsWith("mesos://") &&
```

Comment:
You could maybe refactor the check into a small private method to avoid duplication, but no big deal

Reply:
ok